api/v1/services/StatsService.js

```diff
@@ -0,0 +1,137 @@
+var _Promise = require('bluebird');
+var _ = require('lodash');
+
+var database = require('../../database');
+
+var Attendee = require('../models/Attendee');
+var AttendeeEcosystemInterest = require('../models/AttendeeEcosystemInterest')
+var Ecosystem = require('../models/Ecosystem');
+var User = require('../models/User');
+
+var errors = require('../errors');
+var utils = require('../utils');
+var cache = utils.cache;
+
+
+/**
+* Fetches the current stats, requerying them if not cached
+* @return {Promise<Object>}	resolving to key-value pairs of stats
+*/
+module.exports.fetchStats = function () {
+    return cache.hasKey('stats')
+    .then(function(hasKey){
+        if(hasKey) {
+            return cache.getString('stats')
+            .then(function(object){
+                return JSON.parse(object);
+            });
+        }
+        else {
+            var stats = {}
+            var populateStats = function(key) {
```

Comment:
Does this have to be declared in here or can we have something like this declared as a helper function towards the top:
```
function populateStats(key, stats) {
   return function (result) { ... }
}
```